### PR TITLE
Post new Sleep / Activity / Hydration data

### PR DESCRIPTION
**Pull request by Karlfunhouse**
# PR Description
Please include a summary of the change:
Adding functionality in for:
-postNewSleepData()
-postNewActivityData()
-postNewHydrationData()

Commit Emojis:
:crystal_ball: JS modifications/updates.

If it is a new feature, describe the feature and the iteration of the project spec it addresses.
Adding functionality in to get user input data from Data Input form.  
POST new entries to corresponding datasets.

Note any questions or struggles as well.

## Type of change
Please delete options that are not relevant.
- [ ] New feature (non-breaking change which adds functionality)

## Checklist:
- [ ] My code follows the Turing style guidelines
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -22,7 +22,7 @@ <h3 class='dropdown-category'>Sleep</h3>
             <input type='number' class='hours-slept-input-js data-input-field' required min='0' max='24'></input>
             <p class='dropdown-p black'>Sleep Quality</p>
             <input type='number' class='sleep-quality-input-js data-input-field' required min='0' max='5'></input>
-            <button type='button' name='button' class='add-activity-button-js'>Add Sleep Data</button>
```
Comment (Karlfunhouse) [inline src/index.html:25]: updating appropriate class names

File: src/scripts.js [hunk 2]
```diff
@@ -62,6 +63,10 @@ Promise.all([userData, sleepData, activityData, hydrationData])
     displayUserInfo(user, todayDate);
     // sortedHydrationDataByDate(user);
   })
+  .then(() => {
```
Comment (Karlfunhouse) [inline src/scripts.js:66]: DELETE THIS

File: src/scripts.js [hunk 3]
```diff
@@ -109,6 +114,9 @@ $('#profile-button').on('click', (event) => showUserDropdown());
 $('#add-data-button').on('click', (event) => showActivityDropdown());
 $('.stairs-trending-button').on('click', (event) => updateTrendingStairsDays)
 $('.steps-trending-button').on('click', (event) => updateTrendingStepDays)
+$('.add-sleep-button-js').on('click', (event) => postNewSleepData())
```
Comment (Karlfunhouse) [inline src/scripts.js:117]: Adding event listeners to add new data buttons

File: src/scripts.js [hunk 4]
```diff
@@ -128,6 +136,74 @@ function showActivityDropdown() {
 // have to create four different jQuery event listeners,
 // so unsure which option is better.
 
+
+
+function postNewSleepData() {
```
Comment (Karlfunhouse) [inline src/scripts.js:141]: Post New Sleep Data function

File: src/scripts.js [hunk 5]
```diff
@@ -128,6 +136,74 @@ function showActivityDropdown() {
...
+  .then((response) => response.json())
+  .then((data) => {
+    console.log('Success:', data);
+  })
+  .catch((error) => {
+    console.error('Error:', error);
+  });
+};
+
+function postNewActivityData() {
```
Comment (Karlfunhouse) [inline src/scripts.js:163]: function to post new activity data

File: src/scripts.js [hunk 6]
```diff
@@ -128,6 +136,74 @@ function showActivityDropdown() {
...
+  .then((response) => response.json())
+  .then((data) => {
+    console.log('Success:', data);
+  })
+  .catch((error) => {
+    console.error('Error:', error);
+  });
+};
+
+function postNewHydrationData() {
```
Comment (Karlfunhouse) [inline src/scripts.js:186]: function to post new hydration data to server

File: src/scripts.js [hunk 7]
```diff
@@ -389,68 +468,8 @@ function displaySleepData(user, todayDate) {
 // };
 // //splice sortedHydrationDataByDate to get most recent 7 entires
 // //loop through the first 7 entries and populate the oz per day
-
```
Comment (Karlfunhouse) [inline src/scripts.js:392]: deleting repetitive code

File: src/scripts.js [hunk 8]
```diff
@@ -11,6 +11,7 @@ import $ from 'jquery';
 // GLOBALS
 let userRepository;
 let userData;
+let user;
```
Comment (collinkallery) [inline src/scripts.js:14]: do we need this still?

File: src/index.html (hunk 1 above)
Comment (Melizzo) [reply src/index.html:25]: All members present - Karl really rocked out the posting.  Good lessons learned on ensuring the id is exactly the same as the dataset

File: src/scripts.js [hunk 9]
```diff
@@ -128,6 +136,74 @@ function showActivityDropdown() {
...
+    body: JSON.stringify({
+      "userID": Number($('.user-id-js').text()),
+      "date": $('.date-input').val().split('-').join('/'),
+      "hoursSlept": Number($('.hours-slept-input-js').val()),
+      "sleepQuality": Number($('.sleep-quality-input-js').val())
+    })
+  })
+  .then((response) => response.json())
+  .then((data) => {
+    console.log('Success:', data);
```
Comment (collinkallery) [inline src/scripts.js:156]: also do we need this success console log?

**Pull request merged**